### DISCUSSION TITLE
How to hydrate a machine asynchronously?

### DISCUSSION THREAD

**Discussion by codingedgar** [statelyai/xstate#2331] [Q&A]
Hi!

I'm using `react-native` with `xstate` and want to do some persistence in the docs (https://xstate.js.org/docs/packages/xstate-react/#persisted-and-rehydrated-state), the reading is done synchronously, thus not dealing with async get/set, which is the case for `@react-native-async-storage/async-storage`.

Right now, I'm dealing with it by just saving the context of the machine and hydrating it on an initial state, not the whole state, more concretely:

How to hydrate a machine asynchronously with all the features from the `UseMachine.options[state]`

```ts
  const [state, send] = useMachine(someMachine, {
    state: persistedState // provide persisted state config object here
  });
```

**Comment by davidkpiano** [comment]
You can use a wrapper component to deal with this:

```js
const storage = useGetAsyncStorage(); // implementation left as exercise to the reader

if (!storage) {
  return null;
}

return <Component state={storage.persistedState} />
```
